utils/src/main/scala/it/pagopa/interop/commons/utils/OpenapiUtils.scala

```diff
@@ -19,19 +20,27 @@ trait OpenapiUtils {
     params.isEmpty || params.contains(s)
   }
 
-  def errorFromRequestValidationReport(report: ValidationReport): String = {
-    val messageStrings = report.getMessages.asScala.foldLeft[List[String]](List.empty)((tail, m) => {
-      val context = m.getContext.toScala.map(c =>
-        Seq(c.getRequestMethod.toScala, c.getRequestPath.toScala, c.getLocation.toScala).flatten
+  def errorFromRequestValidationReport(report: ValidationReport): List[ValidationRequestError] = {
+    val errors: List[ValidationRequestError] = report.getMessages.asScala.toList.map { m =>
+      m.getContext.toScala.flatMap(_.getParameter.toScala) match {
```

Comment:
how about a fold? a `.map(...).getOrElse`? Please don't pattern match over option 🙏

Reply:
Typically I prefer to use `fold` o `map/getOrElse`, but in this specific case, I chose pattern matching because it looks more readable.
The solution with `map/getOrElse` is still readable: I'm moving to that.